lib/vtls/openssl.c

```diff
@@ -3984,13 +3986,27 @@ static CURLcode ossl_connect_step2(struct Curl_cfilter *cf,
     }
   }
   else {
+    int psigtype_nid;
+    const char *negotiated_group_name;
+
     /* we connected fine, we're not waiting for anything else. */
     connssl->connecting_state = ssl_connect_3;
 
     /* Informational message */
-    infof(data, "SSL connection using %s / %s",
+    psigtype_nid = NID_undef;
+    SSL_get_peer_signature_type_nid(backend->handle, &psigtype_nid);
+#if defined(OPENSSL_VERSION_MAJOR) && defined(OPENSSL_VERSION_MINOR) && \
+    (OPENSSL_VERSION_MAJOR >= 3) && (OPENSSL_VERSION_MINOR >= 2)
+      negotiated_group_name = SSL_get0_group_name(backend->handle);
+#elif defined(OPENSSL_VERSION_MAJOR) && (OPENSSL_VERSION_MAJOR >= 3)
```

Comment:
```suggestion
#if (OPENSSL_VERSION_NUMBER >= 0x30200000L)
      negotiated_group_name = SSL_get0_group_name(backend->handle);
#elif (OPENSSL_VERSION_NUMBER >= 0x30000000L)
```
I suggest this shorter method to check for the OpenSSL version, simplifying future extensions for forks. It's the method we use predominantly in `openssl.c`.

Reply:
add in recent commit